cmd/nerdctl/container/container_wait_test.go

```diff
@@ -20,28 +20,31 @@ import (
 	"testing"
 
 	"github.com/containerd/nerdctl/v2/pkg/testutil"
+	"github.com/containerd/nerdctl/v2/pkg/testutil/nerdtest"
+	"github.com/containerd/nerdctl/v2/pkg/testutil/test"
 )
 
 func TestWait(t *testing.T) {
-	t.Parallel()
-	tID := testutil.Identifier(t)
-	testContainerName1 := tID + "-1"
-	testContainerName2 := tID + "-2"
-	testContainerName3 := tID + "-3"
+	testCase := nerdtest.Setup()
 
-	const expected = `0
-0
-123
-`
-	base := testutil.NewBase(t)
-	defer base.Cmd("rm", "-f", testContainerName1, testContainerName2, testContainerName3).Run()
-
-	base.Cmd("run", "-d", "--name", testContainerName1, testutil.CommonImage, "sleep", "1").AssertOK()
+	testCase.Cleanup = func(data test.Data, helpers test.Helpers) {
+		helpers.Anyhow("rm", "-f", data.Identifier("1"), data.Identifier("2"), data.Identifier("3"))
+	}
 
-	base.Cmd("run", "-d", "--name", testContainerName2, testutil.CommonImage, "sleep", "1").AssertOK()
+	testCase.Setup = func(data test.Data, helpers test.Helpers) {
+		helpers.Ensure("run", "-d", "--name", data.Identifier("1"), testutil.CommonImage)
+		helpers.Ensure("run", "-d", "--name", data.Identifier("2"), testutil.CommonImage, "sleep", "1")
+		helpers.Ensure("run", "-d", "--name", data.Identifier("3"), testutil.CommonImage, "sh", "-euxc", "sleep 5; exit 123")
+	}
 
-	base.Cmd("run", "--name", testContainerName3, testutil.CommonImage, "sh", "-euxc", "sleep 5; exit 123").AssertExitCode(123)
+	testCase.Command = func(data test.Data, helpers test.Helpers) test.TestableCommand {
+		return helpers.Command("wait", data.Identifier("1"), data.Identifier("2"), data.Identifier("3"))
```

Comment:
Any reason to call ` data.Identifier("1")` multiple times?

Reply:
In that case, yes (`data.Identifier()` is tied to the current `testCase` - ensuring sub-tests, or parallel tests, can never collide id.

Still good feedback though - for future evolutions of the test toolkit.